ttnn/cpp/ttnn/operations/data_movement/common/common.cpp

```diff
@@ -55,6 +55,84 @@ ttnn::Tensor pad_to_tile_vol(
     return tensor;
 }
 uint32_t wrap_index(int index, int size) { return index < 0 ? size + index : index; }
+
+ttnn::MemoryConfig create_sharded_memory_config(
+    const ttnn::Shape& shape,
+    const tt::tt_metal::CoreRangeSet& core_grid,
+    const ShardStrategy& strategy,
+    const tt::tt_metal::ShardOrientation& orientation,
+    bool halo,
+    bool use_height_and_width_as_shard_shape,
+    const tt::tt_metal::Layout& layout) {
+    auto is_tile_layout = layout == tt::tt_metal::Layout::TILE;
+
+    auto rank = shape.rank();
+    TT_FATAL(rank >= 2, "rank of tensor to shard must be at least 2.");
```

Comment:
Is this an actual requirement? Can't we shard a tensor with 0 or 1 dimensions?
I think if you remove the check it should work just fine.

Reply:
I did a brain-off translation from the Python version, so you're probably right. I'll give it a shot tomorrow.

Reply:
so, can this check be removed?

Reply:
I didn't get a chance to test it out on a 0 or 1 dim tensor. Since we're not using it for 0 or 1 dimension tensors at this time, I propose that we keep the check in place and revisit the subject if the function is used more broadly (this is currently common code only for data movement and only used for rank 4 tensors).